Automattic/wp-calypso
# Editor mentions: share suggestion components and styles with user-mentions block

To reduce duplication of components and styles, this PR points the Editor Mentions plugin at the new `user-mentions` block for the following:
- `SuggestionList` component
- `Suggestion` component
-  styles for the above components

Fixes #24642.

### To test

Load up the Editor by creating a new post for a site with lots of users (like a P2). Ensure that there are no visual or functional regressions when you type '@' to choose a user.

<img width="324" alt="screen shot 2018-05-04 at 4 05 21 pm" src="https://user-images.githubusercontent.com/17325/39612727-fdbc7d88-4fb5-11e8-852f-f7d59a311c8b.png">
